### PR TITLE
Attribution and Edition features

### PR DESCRIPTION
**Pull request by bdenham**
## Description

This PR adds both Edition and Attribution components to satisfy the documentation requirements for Commerce DevDocs, as specified in these two Jira tickets:

- https://jira.corp.adobe.com/browse/COMDOX-76
- https://jira.corp.adobe.com/browse/COMDOX-77

## Motivation and Context

These enhancements are necessary to provide features for editions and partner attributes within Commerce devdocs.

## How Has This Been Tested?

Local testing with frontmatter that matches the frontmatter for these features in devdocs pages. 

## Screenshots (if appropriate):

image

## How to Test

image

1. Pull down the branch locally.
2. Open a terminal in the root directory.
3. Run `yarn` or `yarn install` (same thing) to download and install the packages.
4. Run `yarn dev` to build the theme and the example site.
5. If the build succeeds, go here: http://localhost:8000/support/FAQ/
6. You should see something like the screen shot above.
7. Now navigate to the FAQ page from the project root: `example/src/pages/support/FAQ/index.md`
8. Change any of the following frontmatter: 
     `contributor_name: Kalpesh Mehta from Corra`
     `contributor_link: https://partners.magento.com/portal/details/partner/id/70/`
     `edition: ee`
9. With each change, you need to save the file, stop your local server, and rebuild the project by runing `yarn dev` again to see the FAQ page changes.

The reason you need to stop the server and rebuild between each change is because the GraphQL query that retrieves the frontmatter data only runs once per build and stored for the application to use. So if you change the frontmatter, you need to rerun the query—rebuild the project.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: example/src/pages/support/FAQ/index.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+---
+title: FAQ
+description: This is the FAQ page for Adobe Analytics
+contributor_name: Kalpesh Mehta from Corra
+contributor_link: https://partners.magento.com/portal/details/partner/id/70/
+edition: ee_only
```
Comment (jeff-matthews) [inline example/src/pages/support/FAQ/index.md:6]: Per conversation in grooming today, let's drop "_only" from all options.

**Review by jeff-matthews — COMMENTED** [review]
Looking good @bdenham! I'm excited to see these enhancements. The attribution is much more prominent and easier to see.

This might be a question for @timkim and @dmitrymatio, but are the proposed `contributor_name:` and `contributor_ link:` properties and components replacing the existing `contributor:` property and component?

File: example/src/pages/support/FAQ/index.md (hunk 1 above)
Comment (timkim) [reply example/src/pages/support/FAQ/index.md:6]: Hmm I'm not certain how to tie in the github info if it's defined in the frontmatter portion. Currently the info is loaded in via  plugin (https://github.com/adobe/gatsby-source-github-file-contributors) then passed to Provider (https://github.com/adobe/aio-theme/blob/main/packages/gatsby-theme-aio/src/components/Layout/index.js#L185) and then injected into the page (https://github.com/adobe/aio-theme/blob/3f0a6fa59ae62314d9c72ff04044d9ad9fda56b6/packages/gatsby-theme-aio/src/components/MDXFilter/index.js#L348).

If I recall our conversation correctly, I think you wanted a way to define the contributors more by hand than the current github contributor plugin logic? So I can see having both and then you can reference the one you want.

**Review by jeff-matthews — COMMENTED** [review]
I like the idea of using the AFM extensions so that we can use kramdown inline style syntax for specific use cases similar to how we use them today in our jekyll projects. Ultimately, that depends on whether the Devsite team is willing to adopt AFM.

If that's not a blocker for this PR, we should probably work on that separately.

**Review by jeff-matthews — COMMENTED** [review]
Also, in your last screenshot, I like where you chose to place the labels for the editions. Those  look like inline use cases though, is that right? 

Right now, we only have global use cases (I think).

**Comment by bdenham** [comment]
Yeah, that's right. Those are the inline use cases. And we'll have those for a separate PR as we talked through.

**Review by jeff-matthews — COMMENTED** [review]
It works as expected, although I think you should back out the inline attribute syntax in the example .md files (e.g., `{:.bs-callout-info}`) since they don't apply here. I'm sure they're from some testing you've been doing with integrating AFM. 

Also, I think you should add the new edition component and how to use it to the README.

**Comment by bdenham** [comment]
I added a frontmatter section to the Readme that includes the addition of the new frontmatter `edition` and `contributor_name` / `contributor_link` keywords. This also begs the need to define the other frontmatter keywords, which I have not had the time to do just yet.

**Review by jeff-matthews — APPROVED** [review]
Looks good to me!

**Pull request merged**